include/swift/Syntax/RawSyntax.h

```diff
@@ -300,14 +312,15 @@ class RawSyntax final
   /// Make a raw "layout" syntax node.
   static RC<RawSyntax> make(SyntaxKind Kind, ArrayRef<RC<RawSyntax>> Layout,
                             SourcePresence Presence,
-                            SyntaxArena *Arena = nullptr);
+                            SyntaxArena *Arena = nullptr,
+                            llvm::Optional<unsigned> NodeId = llvm::None);
```

Comment:
If `NodeId` should be always greater than `0`, I think this can be `unsigned NodeId = 0` just like token version. Am I missing something?

Reply:
I initially used 0 to indicate that the NodeId should be picked automatically, but changed it to `Optional<unsigned>` because of @nkcsgexi's comment here: https://github.com/apple/swift/pull/16636#discussion_r198245456. I'm fine with either.

Reply:
I don't think `0` as default is too un-readable. @nkcsgexi WDYT?
Either way, if there's no specific reason, please be consistent between token and layout.